examples/background_tasks.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+"""Example of aiohttp.web.Application.on_startup signal handler"""
+import aioredis
+import asyncio
+from aiohttp.web import Application, run_app, WebSocketResponse
+
+async def websocket_handler(request):
+    ws = WebSocketResponse()
+    await ws.prepare(request)
+    request.app['websockets'].append(ws)
+    try:
+        async for msg in ws:
+            print(msg)
+            await asyncio.sleep(1)
+    finally:
+        request.app['websockets'].remove(ws)
+    return ws
+
+
+async def on_shutdown(app):
+    for ws in app['websockets']:
+        await ws.close(code=999, message='Server shutdown')
+
+
+async def listen_to_redis(app):
+    try:
+        sub = await aioredis.create_redis(('localhost', 6379), loop=app.loop)
+        ch, *_ = await sub.subscribe('news')
+        async for msg in ch.iter(encoding='utf-8'):
+            # Forward message to all connected websockets:
+            for ws in app['websockets']:
+                ws.send_str('{}: {}'.format(ch.name, msg))
+            print("message in {}: {}".format(ch.name, msg))
+    except asyncio.CancelledError:
```

Comment:
`finally` is better than `except CancelledError`.
We should close redis subscription unconditionally.